### PR TITLE
[PR #1537 Patch 3] Fix running examples for text predictor

### PR DESCRIPTION
**Pull request by zhiqiangdon**
*This PR is mainly to fix running examples in `autogluon.examples.text_prediction`*:

1. Fixing passing `presets` from tabular predictor to text predictor.
2. Automatically switching from metric `r2` to `rmse` to avoid a potential bug for per gpu batch 1.
3. Added the cross entropy metric using torchmetrics to match the sklearn's `log_loss`.
4. Removed setting seed in examples since it involves MXNet and setting seed is already done inside AutoMM.
5. Fixed one type hint.

### CODE REVIEW

File: text/src/autogluon/text/automm/optimization/metrics.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import torch
+from torch.nn import functional as F
+from torchmetrics import Metric
+
+
+class CrossEntropy(Metric):
```
Comment (sxjscience) [inline text/src/autogluon/text/automm/optimization/metrics.py:6]: We will need to add a test-case for CrossEntropy. May borrow some example codes from https://github.com/PyTorchLightning/metrics/blob/5cbc7add8877e6524b140eb35f57a26b53fe5670/tests/regression/test_mean_error.py#L88-L140

**Comment by szha** [comment]
Job PR-1565-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1565/2/index.html

**Review by Innixma — APPROVED** [review]

File: text/src/autogluon/text/automm/predictor.py [hunk 2]
```diff
@@ -114,6 +114,10 @@ def __init__(
         if eval_metric is None and problem_type is not None:
             eval_metric = infer_eval_metric(problem_type)
 
+        # Due to torchmetrics, r2 may encounter errors for per gpu batch size 1.
+        if eval_metric is not None and eval_metric.lower() == "r2":
+            eval_metric = "rmse"
```
[the remark this answers is not in the document]
Comment (sxjscience) [reply text/src/autogluon/text/automm/predictor.py:119]: Yeah, this is mainly due to a potential bug in torchmetrics.R2Score when applied in the multi-gpu setting. We can workaround it now by not using R2Score for early stopping.

@zhiqiangdon I think we may still keep "r2" as the eval_metric but just change the early stopping metric at https://github.com/awslabs/autogluon/blob/476dcf3ba8401f791fca84ea48fc9170bd8bafee/text/src/autogluon/text/automm/predictor.py#L358-L361

File: text/src/autogluon/text/automm/optimization/metrics.py (hunk 1 above)
Comment (zhiqiangdon) [reply text/src/autogluon/text/automm/optimization/metrics.py]: Test cases added.

**Review by sxjscience — APPROVED** [review]

**Comment by szha** [comment]
Job PR-1565-3 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1565/3/index.html

**Comment by szha** [comment]
Job PR-1565-4 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1565/4/index.html
